ValveResourceFormat/Resource/ResourceTypes/Material.cs

```diff
@@ -79,6 +73,41 @@ public override void Read(BinaryReader reader, Resource resource)
             {
                 StringAttributes[kvp.GetProperty<string>("m_name")] = kvp.GetProperty<string>("m_pValue");
             }
+
+            // This is zero-length for all vmat files in Dota2 and HL archives
+            var textureAttributes = Data.GetArray<string>("m_textureAttributes");
+            if (textureAttributes.Length > 0) {
+                // Console.WriteLine("unexpected textureAttributes length");
```

Comment:
Should probably be uncommented

Reply:
I wasn't sure what this parameter was so I scanned the dota2 and hl archives for it and found it was 0 in all cases. My guess is that there is a provision for it to do something but because of lack of data I couldn't say anything about what the effects a non-zero value would have. Guessing from the name a non-zero value might point to additional variables. Please confirm what you'd like to do with it

Reply:
Uncomment it, so if it happens we can at least see it and go from there.

Reply:
sounds good